src/inmanta/deploy/scheduler.py

```diff
@@ -532,6 +545,9 @@ async def _new_version(
                 for resource in blocked_resources | transitively_blocked_resources:
                     self._work.delete_resource(resource)
 
+                # Write dirty state back to the database
+                await ResourcePersistentState.mark_as_has_update(self.environment, set(self._state.dirty))
```

Comment:
Not everything in the dirty set has an update. e.g. a repair puts everything in the dirty set, or resources that are `NON_COMPLIANT` (this last one didn't exist yet when you started this PR though).